src/transport_ofi.c

```diff
@@ -87,7 +95,39 @@ static char                     myephostname[EPHOSTNAMELEN];
 #endif
 #ifdef ENABLE_THREADS
 shmem_internal_mutex_t          shmem_transport_ofi_lock;
-#endif
+#ifndef __APPLE__
+#ifdef HAVE_SYS_GETTID
+/* Need a syscall to gettid() because glibc doesn't provide a wrapper
+ * (see gettid manpage in the NOTES section): */
+static inline
+pid_t shmem_transport_ofi_gettid(void)
+{
+    return syscall(SYS_gettid);
+}
+#else
+static inline
+uint64_t shmem_transport_ofi_gettid(void)
+{
+    static uint64_t tid = 0;
+    return tid++;
+}
+#endif /* HAVE_SYS_GETTID */
+#else
+static inline
+uint64_t shmem_transport_ofi_gettid(void)
+{
+    int ret;
+    uint64_t tid;
+    ret = pthread_threadid_np(NULL, &tid);
+    if (ret == 0) {
+      return tid;
+    } else {
+        RAISE_ERROR_MSG("Error geting thread ID: %s \n", strerror(ret));
+        return -1;
```

Comment:
Do you want to return -1 or abort?  ```RAISE_ERROR_*``` will abort (and annotated with noreturn), so the return should be unreachable.  Also, you can't return -1 in a uint.

Reply:
Yeah, I just want to abort.